ledger/src/blockstore/blockstore_purge.rs

```diff
@@ -36,16 +54,26 @@ impl Blockstore {
     /// while the non-slot-id based column families, `cf::TransactionStatus`,
     /// `AddressSignature`, and `cf::TransactionStatusIndex`, are cleaned-up
     /// based on the `purge_type` setting.
-    pub fn purge_slots(&self, from_slot: Slot, to_slot: Slot, purge_type: PurgeType) {
+    pub fn purge_slots(&self, purge_ranges: PurgeRanges, purge_type: PurgeType) {
```

Comment:
I made this function take `PurgeRanges` which could consist of multiple ranges so that the case where we're clearing some range of coding shreds and some range of all columns can use the same `WriteBatch` and only need to grab `w_active_transaction_status_index` once. Doing each of these once should be minimize total overhead vs. having to do each of these items in two separate functions calls; however, it isn't super pretty ...

As such, I left the other purge functions as is for the moment. If `PurgeRanges` seems reasonable, I can update those as well.